File: 48-rotate-image/48-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int n=matrix.length;
        //swap rows
        for(int i=0;i<n/2;i++){
            int[] row=matrix[i];
            matrix[i]=matrix[n-1-i];
            matrix[n-1-i]=row;
        }
        
        //tranpose
        for(int i=0;i<n;i++){
            for(int j=i;j<n;j++){
                int ele=matrix[j][i];
                matrix[j][i]=matrix[i][j];
                matrix[i][j]=ele;
            }
        }
    }
}

File: 48-rotate-image/48-Rotate-Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& mat) {
        int n = mat.size();
        int m = mat[0].size();
        for(int i=0;i<n;i++){
            for(int j=i;j<m;j++){
                int t = mat[i][j];                
                mat[i][j] = mat[j][i];
                mat[j][i] = t;
            }
        }
        for(int i=0;i<n;i++){
            reverse(mat[i].begin(), mat[i].end());
        }        
    }
};
